src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -457,10 +458,20 @@ void ov::npuw::JustInferRequest::unpack_closure(std::size_t idx, RqPtr request)
             // Remember where the unpack is required
             closure_unpack_required.push_back(cidx);
         } else {
-            // Easy case, just set one to another. Copy_to is also possible
-            // and even may be preferrable for some devices, like this:
-            // ```ov::get_tensor_impl(closure)->copy_to(clparam._ptr);'''
-            request->set_tensor(iport, ov::get_tensor_impl(closure));
+            // Utilize weights bank if applicable
+            if (!m_npuw_model->m_weights_bank_name.empty()) {
+                auto weights_bank =
+                    ov::npuw::weights_bank::WeightsBankManager::getInstance().getBank(m_npuw_model->m_weights_bank_name,
+                                                                                      *comp_model_desc.device_it);
+                auto remote_tensor =
+                    weights_bank->getSharedTensor(closure.get_element_type(), closure.get_shape(), closure.data());
```

Comment:
why not pass the whole `ov::Tensor`? Also, I hope this is not the final implementation, just a sketch.

We expect weight tensors to be specified only once, at the load time. You don't do it during the execution.

Reply:
Done